src/ledger/reader.zig

```diff
@@ -515,7 +520,11 @@ pub const BlockstoreReader = struct {
             const signature = transaction.signatures[0];
             txns_with_statuses.appendAssumeCapacity(.{
                 .transaction = transaction,
-                .meta = try self.db.get(schema.transaction_status, .{ signature, slot }) orelse
+                .meta = try self.db.get(
+                    schema.transaction_status,
+                    .{ signature, slot },
+                    self.allocator,
```

Comment:
With these methods, it might be best to update to the sig convention of passing allocators as the first non-self argument to function calls unless there is a compelling reason not to

Reply:
f59c2103b90928234a562f530f0164ab957886e0